isis/src/base/apps/isis2std/isis2std.cpp

```diff
@@ -0,0 +1,145 @@
+/** This is free and unencumbered software released into the public domain.
+
+The authors of ISIS do not claim copyright on the contents of this file.
+For more details about the LICENSE terms and the AUTHORS, you will
+find files of those names at the top level of this repository. **/
+
+/* SPDX-License-Identifier: CC0-1.0 */
+
+#include "ExportDescription.h"
+#include "FileName.h"
+#include "ImageExporter.h"
+#include "UserInterface.h"
+
+
+using namespace std;
+
+namespace Isis  {
+
+  int addChannel(UserInterface &ui, ExportDescription &desc, QString param, QString mode);
+  void addResults(PvlGroup &results, ImageExporter *exporter, QString channel, int index);
+
+
+  void isis2std(UserInterface &ui, Pvl *log) {
+    QString format = ui.GetString("FORMAT");
+    ImageExporter *exporter = ImageExporter::fromFormat(format);
+
+    ExportDescription desc;
+    if (ui.GetString("BITTYPE") == "8BIT") {
+      desc.setPixelType(UnsignedByte);
+    }
+    else if (ui.GetString("BITTYPE") == "U16BIT") {
+      desc.setPixelType(UnsignedWord);
+    }
+    else { //if (ui.GetString("BITTYPE") == "S16BIT")
+      desc.setPixelType(SignedWord);
+    }
+
+    int redIndex = -1;
+    int greenIndex = -1;
+    int blueIndex = -1;
+    int alphaIndex = -1;
+
+    QString mode = ui.GetString("MODE");
+    if (mode == "GRAYSCALE") {
+      addChannel(ui, desc, "FROM", mode);
+      exporter->setGrayscale(desc);
+    }
+    else {
+      redIndex = addChannel(ui, desc, "RED", mode);
+      greenIndex = addChannel(ui, desc, "GREEN", mode);
+      blueIndex = addChannel(ui, desc, "BLUE", mode);
+
+      if (mode == "ARGB") {
+        alphaIndex = addChannel(ui, desc, "ALPHA", mode);
+        exporter->setRgba(desc);
+      }
+
+      else {
+        exporter->setRgb(desc);
+      }
+    }
+
+    FileName outputName = ui.GetFileName("TO");
+    int quality = ui.GetInteger("QUALITY");
+
+
+    QString compression;
+    if (format == "TIFF") {
+      compression = ui.GetString("COMPRESSION").toLower();
```

Comment:
Do we need to perform any validation on the compression input, or is that handled elsewhere?

Reply:
std2isis verifies the compressed output when it re-ingests the image.